Entities/Items/Crate/Crate.as

```diff
@@ -346,10 +346,32 @@ void onCommand(CBlob@ this, u8 cmd, CBitStream @params)
 		}
 		CBlob @caller = getBlobByNetworkID( params.read_u16() );
 
-		if (caller !is null && this.getInventory() !is null) {
-			// We might have to make room
+		if (caller !is null && this.getInventory() !is null) 
+		{
 			CInventory@ inv = this.getInventory();
 			u8 itemcount = inv.getItemsCount();
+
+			// Boobytrap if crate has enemy mine
+			CBlob@ mine = null;
+			for (int i = 0; i < inv.getItemsCount(); i++)
+			{
+				CBlob@ item = inv.getItem(i);
+				if (item.getName() == "mine" && item.getTeamNum() != caller.getTeamNum())
+				{
+					@mine = item;
+					break;
+				}
+			}
+			if (mine !is null)
+			{
+				CBitStream params;
+				params.write_u16(caller.getNetworkID());
+				params.write_u16(mine.getNetworkID());
+				this.SendCommand(this.getCommandID("boobytrap"), params);
+				return;
```

Comment:
Since you return here, you should just move this inside the loop at line 361.

Reply:
I've also edited the part at line ~520 too